src/shared/components/base-authoring.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import Form, { Field, IChangeEvent, UiSchema } from "react-jsonschema-form";
+import { JSONSchema6 } from "json-schema";
+
+import "../../shared/styles/boostrap-3.3.7.css"; // necessary to style react-jsonschema-form
+import css from "../../shared/styles/authoring.scss";
+
+export interface IBaseAuthoringProps<IAuthoredState> {
+  authoredState: IAuthoredState;
+  setAuthoredState?: (state: IAuthoredState) => void;
+  preprocessFormData?: (data: IAuthoredState) => IAuthoredState;
+  // react-jsonschema-form properties:
+  schema: JSONSchema6;
+  uiSchema: UiSchema;
+  // react-jsonschema-form additional fields.
+  fields?: { [name: string]: Field };
+}
+
+export const BaseAuthoring = <IAuthoredState,>({ authoredState, setAuthoredState, preprocessFormData, schema, uiSchema, fields }: IBaseAuthoringProps<IAuthoredState>) => {
```

Comment:
```suggestion
export const BaseAuthoring = <IAuthoredState>({ authoredState, setAuthoredState, preprocessFormData, schema, uiSchema, fields }: IBaseAuthoringProps<IAuthoredState>) => {
```

Reply:
This comma is actually necessary. Otherwise, TS is getting confused - it doesn't know if `<IAuthoredState>` is a generic type or JSX syntax. ;) If there are two types provided, it's already fine.